src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpoint.cs

```diff
@@ -26,6 +21,16 @@ public class ServiceEndpoint
 
         internal int? Port { get; }
 
+        /// <summary>
+        /// When there are server connections connected to the target endpoint for current hub, the endpoint is considered as Online; otherwise, Offline.
+        /// </summary>
+        public bool Online { get; internal set; } = true;
```

Comment:
Can we call it something like `HasServerConnected`? `Online` looks like describing the service is available or not

Reply:
It is public and hard to change. and I think Online means the service is available to this app server instance which sounds ok to me.

Reply:
And when all server connections are offline, the service looks unreachable to the server, it is offline.

Reply:
add more comments